cal/cal/static/js/services.js

```diff
@@ -220,24 +220,69 @@ analyticsApp.service('QueryService', function() {
     var ctrlDetails = [];
     var maxYValue = 0;
     var events = [];
+    var xLabels = [];
+    var yLabels = [];
     for (var i = 0; i < data.length; i++) {
       var event = data[i];
       var date = new Date(event[0]);
       var hours = event[1];
       if (hours > maxYValue) {
         maxYValue = hours;
       };
+      xLabels.push(date);
+      yLabels.push(hours);
       events.push({
         x: date,
         y: hours
       });
     };
-    ctrlDetails = [{
+    var xSeries = d3.range(1, xLabels.length + 1);
+    var leastSquaresCoeff = leastSquares(xSeries, yLabels);
+
+    // apply the results of the least squares regression
+    var x1 = xLabels[0];
+    var y1 = leastSquaresCoeff[0] + leastSquaresCoeff[1];
+    var x2 = xLabels[xLabels.length - 1];
+    var y2 = leastSquaresCoeff[0] * xSeries.length + leastSquaresCoeff[1];
+    var trendData = [{x:x1,y:y1},{x:x2,y:y2}];
+
+    ctrlDetails.push({
       values: events,
       key: type + ' Graph',
       color: '#003057',
-      strokeWidth: 2,
-    }];
+      strokeWidth: 1,
+    });
+
+    ctrlDetails.push({
+      values: trendData,
+      key: 'Trend Line',
+      color: '#FDB515',
+      strokeWidth: 3,
+    });
+
     return [ctrlDetails, maxYValue];
   };
 });
+
+// returns slope, intercept and r-square of the line
+function leastSquares(xSeries, ySeries) {
```

Comment:
You can have another function in QueryService and call it by doing QueryService.leastSquares. You'll also have to make sure whatever controller is using this function takes in QueryService as a controller in its component